ddtrace/internal/forksafe.py

```diff
@@ -97,15 +100,23 @@ def _reset_object(self):
 
 
 def Lock():
-    # type: (...) -> ResetObject
+    # type: (...) -> ResetObject[threading.Lock]
     return ResetObject(threading.Lock)
 
 
 def RLock():
-    # type: (...) -> ResetObject
+    # type: (...) -> ResetObject[threading.RLock]
     return ResetObject(threading.RLock)
 
 
+def Condition():
+    # type: (...) -> threading.Condition
+    return threading.Condition(lock=RLock())
+
+
 def Event():
-    # type: (...) -> ResetObject
-    return ResetObject(threading.Event)
+    # type: (...) -> threading.Event
+    event = threading.Event()
+    # Replace internal Condition by a forksafe version
+    event._cond = Condition()  # type: ignore[attr-defined]
```

Comment:
Would it be safer to use something like a context manager to patch `threading.Condition` instead of relying on the "private" `_cond`?

Reply:
I'm not sure I understand what you mean by a context manager here?

Reply:
~~~ python
def Event():
    with patched_condition:
        return threading.Event()
~~~

Reply:
Oh. I wouldn't bother. Using the "private" attribute works well enough on every Python version and I don't expect CPython to change this every release. 🤷